engine/src/flutter/shell/platform/darwin/ios/framework/Source/FlutterSceneDelegate.m

```diff
@@ -0,0 +1,29 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import "flutter/shell/platform/darwin/ios/framework/Headers/FlutterSceneDelegate.h"
+
+#import "flutter/shell/platform/darwin/common/framework/Headers/FlutterMacros.h"
+#import "flutter/shell/platform/darwin/ios/framework/Source/FlutterSharedApplication.h"
+
+FLUTTER_ASSERT_ARC
+
+@implementation FlutterSceneDelegate
+
+- (void)scene:(UIScene*)scene
+    willConnectToSession:(UISceneSession*)session
+                 options:(UISceneConnectionOptions*)connectionOptions {
+  NSObject<UIApplicationDelegate>* appDelegate = FlutterSharedApplication.application.delegate;
+  if (appDelegate.window.rootViewController) {
+    // If this is not nil we are running into a case where someone is manually
+    // performing root view controller setup in the UIApplicationDelegate.
```

Comment:
Let's print a warning here. After adopting Scene Delegate API, even if they programmatically setup VC, they should do it in SceneDelegate, not AppDelegate.

Reply:
Done.